client/src/fabric/features.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2019. Beautiful Code BV, Rotterdam, Netherlands
+ * Licensed under GNU GENERAL PUBLIC LICENSE Version 3.
+ */
+
+import { BehaviorSubject } from "rxjs"
+
+import { IFabricEngine, IntervalRole, PhysicsFeature } from "./fabric-engine"
+import { FabricInstance } from "./fabric-kernel"
+
+export interface IFeatureName {
+    physicsFeature?: PhysicsFeature
+    intervalRole?: IntervalRole
+}
+
+function nameLabel(feature: IFeatureName): string {
+    if (feature.physicsFeature !== undefined) {
+        return PhysicsFeature[feature.physicsFeature]
+    }
+    if (feature.intervalRole !== undefined) {
+        return IntervalRole[feature.intervalRole]
+    }
+    return "?"
+}
+
+export interface IFeature {
+    label: string
+    name: IFeatureName
+    isPhysics: boolean
+    setFactor: (factor: number) => void
+    factor$: BehaviorSubject<number>
+    defaultValue: number
+}
+
+export function applyFeatureToInstance(instance: FabricInstance):  (feature: IFeature) => void {
+    return feature => {
+        const intervalRole = feature.name.intervalRole
+        if (intervalRole === undefined) {
+            throw new Error()
```

Comment:
Throw in an error message for the poor developers

Reply:
Yes.